codes/c/chapter_backtracking/preorder_traversal_i_compact.c

```diff
@@ -0,0 +1,85 @@
+/**
+ * File: preorder_traversal_i_compact.c
```

Comment:
I recommend that create another PR for the chapter backtracking. It is more easily to trace the PR.